src/netbox_client.py

```diff
@@ -0,0 +1,186 @@
+import http
+import logging
+from json import JSONDecodeError
+
+import requests
+from requests import Session, exceptions
+
+
+class NboxRecord:
+    """
+    Generic NetBox record class represents Netbox entity
+    """
+    __slots__ = ("_record", "_type", "name", "nsg_tags", "id", "device_id")
+
+    def __init__(self, record: dict):
+        self._record = record
+        url: str = record.get("url")
+        if url:
+            parts = url.strip("/").rsplit("/", maxsplit=2)
+            if len(parts) == 3 and parts[2].isdigit():
+                self._type = parts[1].lower()
+        else:
+            self._type = None
+        self.name = record.get("name")
+        self.id = record.get("id")
+        self.nsg_tags = {}
+        self.device_id = record.get("device", {}).get("id")  # if hasattr(record, "device") else None
+
+    def __getattr__(self, item):
+        res = self._record.get(item)
+        if isinstance(res, dict):
+            if len(res) == 0:
+                return None
+            return NboxRecord(res)
+        if isinstance(res, (tuple,list)):
+            result = []
+            for item_ in res:
+                if isinstance(item_, dict):
+                    result.append(NboxRecord(item_))
+                else:
+                    result.append(item_)
+            return result
+        return res
+
+    def __str__(self):
+        return self._record.get("name",
+                                self._record.get("label", self._record.get("display", self._record.get("slug", ""))))
+
+    def __repr__(self):
+        return self._record.get("name",
+                                self._record.get("label", self._record.get("slug", self._record.get("display")))) \
+            or "unknown"
+
+    def __format__(self, fmt):
+        return format(self.__str__(), fmt)
+
+
+class NboxDevice(NboxRecord):
+    """
+    Netbox Device record
+    """
+    __slots__ = ("iFaces_tags",  "primary_ip", "interface_count")
+
+    def __init__(self, device: dict):  # pynetbox.core.response.Record):
+        super().__init__(device)
+        self.device_id = self.id
+        self.iFaces_tags = {}
+        self.interface_count = device.get('interface_count')
+        self.primary_ip = None
+
+
+class NetboxClient:
```

Comment:
If I am not mistaken, you used to use on some Netbox client library but not you have dropped it. I am curious, why ?